js/src/main/scala/java/util/TimeZone.scala

```diff
@@ -84,7 +92,22 @@ abstract class TimeZone extends Serializable with Cloneable {
   def getDisplayName(daylight: Boolean, style: Int, locale: Locale): String = {
     if (style != TimeZone.SHORT || style != TimeZone.LONG)
       throw new IllegalArgumentException(s"Illegal timezone style: $style")
-    ???
+
+    val id = getID
+    val zoneName = DateFormatSymbols.getInstance(locale).getZoneStrings.find(_(0) == id).flatMap { strs =>
```

Comment:
This is nice. I don't think we have this data en `scala-java-locales` but if we had it it would be good to get it